pkg/agent/openflow/pipeline.go

```diff
@@ -118,114 +118,92 @@ func (a ofAction) String() string {
-		{EgressMetricTable, "EgressMetric"},
-		{l3ForwardingTable, "L3Forwarding"},
-		{snatTable, "SNAT"},
-		{l3DecTTLTable, "IPTTLDec"},
-		{l2ForwardingCalcTable, "L2Forwarding"},
-		{AntreaPolicyIngressRuleTable, "AntreaPolicyIngressRule"},
-		{IngressRuleTable, "IngressRule"},
-		{IngressDefaultTable, "IngressDefaultRule"},
-		{IngressMetricTable, "IngressMetric"},
-		{conntrackCommitTable, "ConntrackCommit"},
-		{serviceConntrackCommitTable, "ServiceConntrackCommit"},
-		{hairpinSNATTable, "HairpinSNATTable"},
-		{L2ForwardingOutTable, "Output"},
+	egressTables = map[uint8]struct{}{
+		AntreaPolicyEgressRuleTable.GetID(): {},
+		EgressRuleTable.GetID():             {},
+		EgressDefaultTable.GetID():          {},
 	}
+
+	// ofTableCache caches the OpenFlow tables used in the pipeline, and it supports using the table ID and name as the index to query the OpenFlow table.
+	ofTableCache = cache.NewIndexer(tableIDKeyFunc, cache.Indexers{tableNameIndex: tableNameIndexFunc})
 )
 
-// GetFlowTableName returns the flow table name given the table number. An empty
+func tableNameIndexFunc(obj interface{}) ([]string, error) {
+	ofTable := obj.(binding.Table)
+	return []string{ofTable.GetName()}, nil
+}
+
+func tableIDKeyFunc(obj interface{}) (string, error) {
+	ofTable := obj.(binding.Table)
+	return fmt.Sprintf("%d", ofTable.GetID()), nil
+}
+
+func getTableByID(id uint8) binding.Table {
+	obj, exists, _ := ofTableCache.GetByKey(fmt.Sprintf("%d", id))
+	if !exists {
+		return nil
+	}
+	return obj.(binding.Table)
+}
+
+// GetFlowTableName returns the flow table name given the table ID. An empty
 // string is returned if the table cannot be found.
-func GetFlowTableName(tableNumber binding.TableIDType) string {
-	for _, t := range FlowTables {
-		if t.Number == tableNumber {
-			return t.Name
-		}
+func GetFlowTableName(tableID uint8) string {
+	table := getTableByID(tableID)
+	if table == nil {
+		return ""
 	}
-	return ""
+	return table.GetName()
 }
 
-// GetFlowTableNumber does a case insensitive lookup of the table name, and
+// GetFlowTableID does a case insensitive lookup of the table name, and
 // returns the flow table number if the table is found. Otherwise TableIDAll is
 // returned if the table cannot be found.
-func GetFlowTableNumber(tableName string) binding.TableIDType {
-	for _, t := range FlowTables {
-		if strings.EqualFold(t.Name, tableName) {
-			return t.Number
-		}
+func GetFlowTableID(tableName string) uint8 {
+	objs, _ := ofTableCache.ByIndex(tableNameIndex, tableName)
+	if len(objs) == 0 {
+		return binding.TableIDAll
 	}
-	return binding.TableIDAll
+	return objs[0].(binding.Table).GetID()
 }
 
-func GetAntreaPolicyEgressTables() []binding.TableIDType {
-	return []binding.TableIDType{
+func PrintTableBaseInfo() string {
+	msg := ""
+	for _, obj := range ofTableCache.List() {
+		t := obj.(binding.Table)
+		msg += fmt.Sprintf("\n  %d\t%s", uint32(t.GetID()), t.GetName())
+	}
+	return msg
+}
```

Comment:
Do you mean we provide function `GetTableList` under `pkg/agent/openflow`?  I want to use a seperate PR to support antctl commands, what do you think?

Reply:
I meant that we should define a `GetTableList` function here instead of `PrintTableBaseInfo`, which is a bit too specific IMO to belong in this package. The `generateFlowTableHelpMsg` function in the antctl code can then call `GetTableList` and format table information as needed.

Reply:
Got it, I would change.